admin_guide/integrations.rst

```diff
@@ -0,0 +1,336 @@
+  * ``/@acceleration/#0`` - the path to element 0.01
+  * ``/@location`` - the path to element ``{"latitude":51.5009449, "longitude":-2.4773414}``
+
+Our own scenario is simple. Our events will be objects like ``{"a":1, "b":2}``. The parser will pull those values into eponymous columns.
+
+Now that the configuration file is set up, it's time to prepare the database. In psql run:
+
+.. code-block:: psql
+
+  -- create metadata tables for kafka-sink-pg-json
+  \i sink/install-justone-kafka-sink-pg-1.0.sql
+
+  -- create and distribute target ingestion table
+  create table kafka_test ( a int, b int );
+  select create_distributed_table('kafka_test', 'a');
+
+Start the Kafka machinery:
+
+.. code-block:: bash
+
+  # save some typing
+  export C=confluent-2.0.0
+
+  # start zookeeper
+  $C/bin/zookeeper-server-start \
+    $C/etc/kafka/zookeeper.properties
+
+  # start kafka server
+  $C/bin/kafka-server-start \
+    $C/etc/kafka/server.properties
+
+  # create the topic we'll be reading/writing
+  $C/bin/kafka-topics --create --zookeeper localhost:2181   \
+                      --replication-factor 1 --partitions 1 \
+                      --topic test
+
+Run the ingestion program:
+
+.. code-block:: bash
+
+  # the jar files for this are in "sink"
+  export CLASSPATH=$PWD/sink/*
+
+  # Watch for new events in topic and insert them
+  $C/bin/connect-standalone \
+    sink/justone-kafka-sink-pg-json-standalone.properties \
+    sink/justone-kafka-sink-pg-json-connector.properties
+
+At this point Kafka-Connect is watching the ``test`` topic, and will parse events there and insert them into ``kafka_test``. Let's send an event from the command line.
+
+.. code-block:: bash
+
+  echo '{"a":42,"b":12}' | \
+    $C/bin/kafka-console-producer --broker-list localhost:9092 --topic test
+
+After a small delay the new row will show up in the database.
+
+::
+
+  select * from kafka_test;
+
+  ┌────┬────┐
+  │ a  │ b  │
+  ├────┼────┤
+  │ 42 │ 12 │
+  └────┴────┘
+
+Caveats
+-------
+
+* At the time of this writing, kafka-sink-pg-json requires Kafka version 0.9 or earlier.
+* The kafka-sink-pg-json connector config file does not provide a way to connect with SSL support, so this tool will not work with Citus Cloud which requires secure connections.
+* A malformed JSON string in the Kafka topic will cause the tool to become stuck. Manual intervention in the topic is required to process more events.
+
+Ingesting Data from Spark
+=========================
+
+People sometimes use Spark to transform Kafka data, such as by adding computed values. In this section we'll see how to ingest Spark dataframes into a distributed Citus table.
+
+First let's start a local Spark cluster. It has several moving parts, so the easiest way is to run the pieces with docker-compose.
+
+.. code-block:: bash
+
+  wget https://raw.githubusercontent.com/gettyimages/docker-spark/master/docker-compose.yml
+  docker-compose up
```

Comment:
This required using `sudo` for me. Else it gives me:
```ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?```

Reply:
OK, it worked on my Mac but I think it has to do with which user the docker daemon runs as. I'll throw a sudo in there for good measure.